src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -9812,6 +9851,311 @@ public UnmanagedCallersOnlyAttribute()
             );
         }
 
+        [Fact]
+        public void UnmanagedCallersOnly_BadExpressionInArguments()
+        {
+
+            var comp = CreateCompilationWithFunctionPointers(new[] { @"
+using System.Runtime.InteropServices;
+class A
+{
+    [UnmanagedCallersOnly(CallConvs = new[] { typeof(Bad, Expression) })]
+    static unsafe void F()
+    {
+        delegate*<void> ptr1 = &F;
+        delegate* unmanaged<void> ptr2 = &F;
+    }
+}
+", UnmanagedCallersOnlyAttribute });
+
+            comp.Assembly.SetOverrideRuntimeSupportsUnmanagedSignatureCallingConvention();
+            comp.VerifyDiagnostics(
+                // (5,54): error CS0246: The type or namespace name 'Bad' could not be found (are you missing a using directive or an assembly reference?)
+                //     [UnmanagedCallersOnly(CallConvs = new[] { typeof(Bad, Expression) })]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Bad", isSuppressed: false).WithArguments("Bad").WithLocation(5, 54),
+                // (5,57): error CS1026: ) expected
+                //     [UnmanagedCallersOnly(CallConvs = new[] { typeof(Bad, Expression) })]
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, ",", isSuppressed: false).WithLocation(5, 57),
+                // (5,59): error CS0103: The name 'Expression' does not exist in the current context
+                //     [UnmanagedCallersOnly(CallConvs = new[] { typeof(Bad, Expression) })]
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "Expression", isSuppressed: false).WithArguments("Expression").WithLocation(5, 59),
+                // (5,69): error CS1003: Syntax error, ',' expected
+                //     [UnmanagedCallersOnly(CallConvs = new[] { typeof(Bad, Expression) })]
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")", isSuppressed: false).WithArguments(",", ")").WithLocation(5, 69),
+                // (8,33): error CS8786: Calling convention of 'A.F()' is not compatible with 'Default'.
+                //         delegate*<void> ptr1 = &F;
+                Diagnostic(ErrorCode.ERR_WrongFuncPtrCallingConvention, "F", isSuppressed: false).WithArguments("A.F()", "Default").WithLocation(8, 33)
+            );
+        }
+
+        [Theory]
+        [InlineData("", 1)]
+        [InlineData("CallConvs = new System.Type[0]", 1)]
```

Comment:
>CallConvs = new System.Type[0]" [](start = 21, length = 31)

Consider testing "CallingConvs = null", "CallingConvs = new Type[] { null }"

Reply:
Will do the former. The latter is covered by `UnmanagedCallersOnlyCallConvNull`.

---
In reply to: [488866675](https://github.com/dotnet/roslyn/pull/47651#discussion_r488866675) [](ancestors = 488866675)